src/libs/migrations/TransactionBackupsToCollection.js

```diff
@@ -0,0 +1,77 @@
+import Onyx from 'react-native-onyx';
+import _ from 'underscore';
+import Log from '@libs/Log';
+import ONYXKEYS from '@src/ONYXKEYS';
+
+// This migration moves all the transaction backups stored in the main transactions collection (ONYXKEYS.COLLECTION.TRANSACTION)
+// to a reserved collection which only stores draft transactions (ONYXKEYS.COLLECTION.TRANSACTION_DRAFT).
+// The main purpose of this migration is that there is a possibility of transactions backups not getting filtered
+// by most functions which need use all transactions. Eg: getAllReportTransactions (src/libs/TransactionUtils.ts)
+// One problem which arose from having duplicate transactions in the collection is that for every distance request
+// which have their waypoints updated offline, we expect the ReportPreview component to display the default image of a pending map.
+// However, due to the presence of a transaction backup, an extra unexpected image of the previous map will be displayed.
+// The problem was further discussed in this PR -> https://github.com/Expensify/App/pull/30232#issuecomment-1781101722
+export default function () {
+    return new Promise((resolve) => {
+        const outerConnectionID = Onyx.connect({
+            key: ONYXKEYS.COLLECTION.TRANSACTION_DRAFT,
+            waitForCollectionCallback: true,
+            callback: (draftTransactions) => {
+                Onyx.disconnect(outerConnectionID);
+
+                // Check if transaction backups were already moved
+                if (draftTransactions) {
```

Comment:
There could be existing drafts already when this migration runs (maybe because my PR is merged for the creation flow). I think it would be better to look for `-backup` transactions in the normal collection in order to do an early return.